### ISSUE TITLE
spaceKLIP failing to run due to numpy 2 just released

### ISSUE THREAD

**Issue by strampelligiovanni** [kammerje/spaceKLIP#186]
I just installed spaceKLIP and I'm getting the following error. 
Looks like a NumPy 2.0.0 and the in the requirements.txt we ask to install the latest Numpy version and that's causing the issue. I solved it on my end putting numpy<2 in the requirments.txt.

```
A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.0.0 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.

Traceback (most recent call last):  File "/Users/gstrampelli/miniconda3/envs/spaceklip/bin/ipython", line 8, in <module>
    sys.exit(start_ipython())
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/__init__.py", line 130, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/traitlets/config/application.py", line 1075, in launch_instance
    app.start()
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/terminal/ipapp.py", line 317, in start
    self.shell.mainloop()
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/terminal/interactiveshell.py", line 917, in mainloop
    self.interact()
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/terminal/interactiveshell.py", line 910, in interact
    self.run_cell(code, store_history=True)
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/core/interactiveshell.py", line 3075, in run_cell
    result = self._run_cell(
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/core/interactiveshell.py", line 3130, in _run_cell
    result = runner(coro)
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/core/async_helpers.py", line 129, in _pseudo_sync_runner
    coro.send(None)
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/core/interactiveshell.py", line 3334, in run_cell_async
    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/core/interactiveshell.py", line 3517, in run_ast_nodes
    if await self.run_code(code, result, async_=asy):
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/core/interactiveshell.py", line 3577, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-1-086efb82d9ed>", line 17, in <module>
    from spaceKLIP import database, coron1pipeline, coron2pipeline, coron3pipeline, pyklippipeline, imagetools, analysistools, plotting
  File "/Users/gstrampelli/spaceKLIP/spaceKLIP/__init__.py", line 1, in <module>
    from . import analysistools
  File "/Users/gstrampelli/spaceKLIP/spaceKLIP/analysistools.py", line 33, in <module>
    from spaceKLIP import utils as ut
  File "/Users/gstrampelli/spaceKLIP/spaceKLIP/utils.py", line 23, in <module>
    from webbpsf_ext.imreg_tools import get_coron_apname as nircam_apname
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/webbpsf_ext/__init__.py", line 100, in <module>
    from .webbpsf_ext_core import MIRI_ext, NIRCam_ext
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/webbpsf_ext/webbpsf_ext_core.py", line 21, in <module>
    from .psfs import nproc_use, gen_image_from_coeff
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/webbpsf_ext/psfs.py", line 7, in <module>
    from .maths import jl_poly
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/webbpsf_ext/maths.py", line 8, in <module>
    from .image_manip import frebin
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/webbpsf_ext/image_manip.py", line 12, in <module>
    import cv2
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/cv2/__init__.py", line 181, in <module>
    bootstrap()
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/cv2/__init__.py", line 153, in bootstrap
    native_module = importlib.import_module("cv2")
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
AttributeError: _ARRAY_API not found

A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.0.0 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.

Traceback (most recent call last):  File "/Users/gstrampelli/miniconda3/envs/spaceklip/bin/ipython", line 8, in <module>
    sys.exit(start_ipython())
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/__init__.py", line 130, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/traitlets/config/application.py", line 1075, in launch_instance
    app.start()
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/terminal/ipapp.py", line 317, in start
    self.shell.mainloop()
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/terminal/interactiveshell.py", line 917, in mainloop
    self.interact()
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/terminal/interactiveshell.py", line 910, in interact
    self.run_cell(code, store_history=True)
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/core/interactiveshell.py", line 3075, in run_cell
    result = self._run_cell(
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/core/interactiveshell.py", line 3130, in _run_cell
    result = runner(coro)
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/core/async_helpers.py", line 129, in _pseudo_sync_runner
    coro.send(None)
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/core/interactiveshell.py", line 3334, in run_cell_async
    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/core/interactiveshell.py", line 3517, in run_ast_nodes
    if await self.run_code(code, result, async_=asy):
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/IPython/core/interactiveshell.py", line 3577, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-1-086efb82d9ed>", line 17, in <module>
    from spaceKLIP import database, coron1pipeline, coron2pipeline, coron3pipeline, pyklippipeline, imagetools, analysistools, plotting
  File "/Users/gstrampelli/spaceKLIP/spaceKLIP/__init__.py", line 1, in <module>
    from . import analysistools
  File "/Users/gstrampelli/spaceKLIP/spaceKLIP/analysistools.py", line 36, in <module>
    from spaceKLIP.pyklippipeline import get_pyklip_filepaths
  File "/Users/gstrampelli/spaceKLIP/spaceKLIP/pyklippipeline.py", line 20, in <module>
    from jwst.pipeline import Detector1Pipeline, Image2Pipeline, Coron3Pipeline
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/jwst/pipeline/__init__.py", line 2, in <module>
    from .calwebb_coron3 import Coron3Pipeline
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/jwst/pipeline/calwebb_coron3.py", line 16, in <module>
    from ..outlier_detection import outlier_detection_step
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/jwst/outlier_detection/__init__.py", line 1, in <module>
    from .outlier_detection_step import OutlierDetectionStep
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/jwst/outlier_detection/outlier_detection_step.py", line 12, in <module>
    from jwst.outlier_detection import outlier_detection
  File "/Users/gstrampelli/miniconda3/envs/spaceklip/lib/python3.11/site-packages/jwst/outlier_detection/outlier_detection.py", line 10, in <module>
    from drizzle.cdrizzle import tblot
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
File ~/miniconda3/envs/spaceklip/lib/python3.11/site-packages/numpy/core/_multiarray_umath.py:44, in __getattr__(attr_name)
     39     # Also print the message (with traceback).  This is because old versions
     40     # of NumPy unfortunately set up the import to replace (and hide) the
     41     # error.  The traceback shouldn't be needed, but e.g. pytest plugins
     42     # seem to swallow it and we should be failing anyway...
     43     sys.stderr.write(msg + tb_msg)
---> 44     raise ImportError(msg)
     46 ret = getattr(_multiarray_umath, attr_name, None)
     47 if ret is None:

ImportError: 
A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.0.0 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.


---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In[1], line 17
     14 import matplotlib.pyplot as plt
     15 import numpy as np
---> 17 from spaceKLIP import database, coron1pipeline, coron2pipeline, coron3pipeline, pyklippipeline, imagetools, analysistools, plotting
     19 # plotting
     20 import matplotlib.pyplot as plt

File ~/spaceKLIP/spaceKLIP/__init__.py:1
----> 1 from . import analysistools
      2 from . import classpsfsubpipeline
      3 from . import coron1pipeline

File ~/spaceKLIP/spaceKLIP/analysistools.py:36
     34 from spaceKLIP.psf import get_offsetpsf, JWST_PSF
     35 from spaceKLIP.starphot import get_stellar_magnitudes, read_spec_file
---> 36 from spaceKLIP.pyklippipeline import get_pyklip_filepaths
     37 from spaceKLIP.utils import write_starfile, set_surrounded_pixels
     39 from webbpsf.constants import JWST_CIRCUMSCRIBED_DIAMETER

File ~/spaceKLIP/spaceKLIP/pyklippipeline.py:20
     17 import pyklip.klip
     19 from astropy import wcs
---> 20 from jwst.pipeline import Detector1Pipeline, Image2Pipeline, Coron3Pipeline
     21 from pyklip import parallelized, rdi
     22 from pyklip.instruments.JWST import JWSTData

File ~/miniconda3/envs/spaceklip/lib/python3.11/site-packages/jwst/pipeline/__init__.py:2
      1 from .calwebb_ami3 import Ami3Pipeline
----> 2 from .calwebb_coron3 import Coron3Pipeline
      3 from .calwebb_dark import DarkPipeline
      4 from .calwebb_detector1 import Detector1Pipeline

File ~/miniconda3/envs/spaceklip/lib/python3.11/site-packages/jwst/pipeline/calwebb_coron3.py:16
     14 from ..coron import align_refs_step
     15 from ..coron import klip_step
---> 16 from ..outlier_detection import outlier_detection_step
     17 from ..resample import resample_step
     19 __all__ = ['Coron3Pipeline']

File ~/miniconda3/envs/spaceklip/lib/python3.11/site-packages/jwst/outlier_detection/__init__.py:1
----> 1 from .outlier_detection_step import OutlierDetectionStep
      2 from .outlier_detection_scaled_step import OutlierDetectionScaledStep
      3 from .outlier_detection_stack_step import OutlierDetectionStackStep

File ~/miniconda3/envs/spaceklip/lib/python3.11/site-packages/jwst/outlier_detection/outlier_detection_step.py:12
      9 from jwst.stpipe import Step
     10 from jwst.lib.pipe_utils import is_tso
---> 12 from jwst.outlier_detection import outlier_detection
     13 from jwst.outlier_detection import outlier_detection_ifu
     14 from jwst.outlier_detection import outlier_detection_spec

File ~/miniconda3/envs/spaceklip/lib/python3.11/site-packages/jwst/outlier_detection/outlier_detection.py:10
      8 from astropy.stats import sigma_clip
      9 from scipy import ndimage
---> 10 from drizzle.cdrizzle import tblot
     12 from stdatamodels.jwst.datamodels.util import open as datamodel_open
     13 from stdatamodels.jwst import datamodels

ImportError: numpy.core.multiarray failed to import
```

**Comment by wbalmer** [comment]
This is also causing all our CI to fail. I opened a PR #188 to fix this for now, but we'll also need to add an issue to upgrade to numpy 2 in the future.

**Comment by AarynnCarter** [comment]
Resolved in PR #188
